.github/workflows/mirror.yaml

```diff
@@ -9,36 +9,95 @@ on:
+          REPO_DIGEST_OF_IMAGE_NEXUS=$(echo "$DOCKER_OUTPUT_NEXUS" | awk '/^\S+: digest: sha256:[0-9a-f]{64} size: [0-9]+$/ { print $3 }');
+          # Check both architectures to see if both are available
+          if [ -z "$REPO_DIGEST_OF_IMAGE_HARBOR" ]; then
+            echo "Could not find repo digest for container image: $DST_REGISTRY_HARBOR/${{ github.event.inputs.harbor_proj }}/${{ github.event.inputs.dst }}-${{ matrix.architecture }}"
+            exit 1
+          fi
+          if [ -z "$REPO_DIGEST_OF_IMAGE_NEXUS" ]; then
+            echo "Could not find repo digest for container image: $DST_REGISTRY_NEXUS/$DST_PROJECT_NEXUS/${{ github.event.inputs.dst }}-${{ matrix.architecture }}"
             exit 1
           fi
           # This generates a signature and publishes it to the registry, next to the image
           # Uses the keyless signing flow with Github Actions as identity provider
-          cosign sign -y "$DST_REGISTRY/$DST_PROJECT/${{ github.event.inputs.dst }}@$REPO_DIGEST_OF_IMAGE"
+          cosign sign -y "$DST_REGISTRY_HARBOR/${{ github.event.inputs.harbor_proj }}/${{ github.event.inputs.dst }}-${{ matrix.architecture }}@$REPO_DIGEST_OF_IMAGE_HARBOR"
+          cosign sign -y "$DST_REGISTRY_NEXUS/$DST_PROJECT_NEXUS/${{ github.event.inputs.dst }}-${{ matrix.architecture }}@$REPO_DIGEST_OF_IMAGE_NEXUS"
+  mirror-multi-arch:
+    name: Create and push manifest list for ${{ github.event.inputs.dst }}
+    runs-on: ubuntu-latest
+    needs: mirror-image
+    permissions:
+      id-token: write
+    steps:
+      - name: Login to Stackable Harbor
+        uses: docker/login-action@0d4c9c5ea7693da7b068278f7b52bda2a190a446 # v3.2.0
+        with:
+          registry: ${{ env.DST_REGISTRY_HARBOR }}
+          username: "robot$sdp+github-action-build"
+          password: ${{ secrets.HARBOR_ROBOT_SDP_GITHUB_ACTION_BUILD_SECRET }}
+      - name: Login to Stackable Nexus
+        uses: docker/login-action@0d4c9c5ea7693da7b068278f7b52bda2a190a446 # v3.2.0
+        with:
+          registry: ${{ env.DST_REGISTRY_NEXUS }}
+          username: github
+          password: ${{ secrets.NEXUS_PASSWORD }}
+      - uses: sigstore/cosign-installer@59acb6260d9c0ba8f4a2f9d9b48431a222b68e20 # v3.5.0
+      - name: Build and push manifest list
+        run: |
+          # Create manifest list for Nexus and Harbor
+          docker manifest create "$DST_REGISTRY_HARBOR/${{ github.event.inputs.harbor_proj }}/${{ github.event.inputs.dst }}" --amend "$DST_REGISTRY_HARBOR/${{ github.event.inputs.harbor_proj }}/${{ github.event.inputs.dst }}-arm64" --amend "$DST_REGISTRY_HARBOR/${{ github.event.inputs.harbor_proj }}/${{ github.event.inputs.dst }}-amd64"
+          docker manifest create "$DST_REGISTRY_NEXUS/$DST_PROJECT_NEXUS/${{ github.event.inputs.dst }}" --amend "$DST_REGISTRY_NEXUS/$DST_PROJECT_NEXUS/${{ github.event.inputs.dst }}-arm64" --amend "$DST_REGISTRY_NEXUS/$DST_PROJECT_NEXUS/${{ github.event.inputs.dst }}-amd64"
```

Comment:
You refer to the images by tag name here, which is not a huge problem, but a race condition might occur:
If the x86 image is built before the ARM image and the build of the ARM image takes a long time, the x86 image might get overridden in the meantime (very unlikely to happen in practice, but possible in theory). That means that the manifest list could potentially point to images with different versions for x86 and ARM.

Since we have the image digests in the build step anyway (needed for signing) you could pass them to the "Create and push manifest list" job as input and then refer to the images by digest when creating the manifest list.

I think it's not a huge problem, just wanted to point it out. If it's not too complicated to fix than I'd prefer that solution since it's the correcter one. If it's too much effort I'm fine with the current solution as well.

Reply:
@dervoeti done with https://github.com/stackabletech/docker-images/pull/738/commits/6c245df3d15b0c25b0906e7a50041f6911121cbb 

thanks